openmc/capi/plot.py

```diff
@@ -0,0 +1,220 @@
+from ctypes import c_int, c_int32, c_double, Structure, POINTER
+
+from . import _dll
+from .core import _DLLGlobal
+from .error import _error_handler
+
+import numpy as np
+
+
+class _Position(Structure):
+    """Definition of an xyz location in space with underlying c-types
+
+    C-type Attributes
+    -----------------
+    x : c_double
+        Position's x value (default: 0.0)
+    y : c_double
+        Position's y value (default: 0.0)
+    z : c_double
+        Position's z value (default: 0.0)
+    """
+    _fields_ = [('x', c_double),
+                ('y', c_double),
+                ('z', c_double)]
+
+    def __getitem__(self, idx):
+        if idx == 0:
+            return self.x
+        elif idx == 1:
+            return self.y
+        elif idx == 2:
+            return self.z
+        else:
+            raise ValueError("{} index is invalid for _Position".format(idx))
```

Comment:
Python's iteration protocol relies on an invalid index raising `IndexError` as opposed to some other exception. If you change this to `raise IndexError`, then you'll be able to do things like:
```Python
r = _Position()
r_list = list(r)
for x in r:
    print(x)
```
With a `ValueError` those wouldn't work.

Reply:
Good to know! I'll try to hang on to that nugget in the future.